pybamm/models/submodels/base_submodel.py

```diff
@@ -70,7 +70,7 @@ def __init__(
         super().__init__(name)
         self.domain = domain
         self.name = name
-
+        self.variables_by_submodel = {}
```

Comment:
can this be deleted?

Reply:
Yes, I just remembered that I need to remove that. Thanks for pointing it out, and thanks for the review, it must've taken a lot of time.